src/main/java/com/coveros/selenified/element/check/State.java

```diff
@@ -0,0 +1,314 @@
+        // record the result
+        if (!isDisplayed) {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + NOTDISPLAYED, timeTook, Success.FAIL);
+        } else {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + DISPLAYED, timeTook, Success.PASS);
+        }
+        return isDisplayed;
+    }
+
+    /**
+     * Verifies that the element is not displayed. If the element isn't present,
+     * it waits up to the default time (5 seconds) for the element, before
+     * marking this verification as a failure. This information will be logged
+     * and recorded, with a screenshot for traceability and added debugging
+     * support.
+     */
+    void notDisplayed();
+
+    default boolean checkNotDisplayed(double waitFor, double timeTook) {
+        // record the action
+        getOutputFile().recordAction(getElement().prettyOutput() + NOTDISPLAYED, waitFor);
+        // verify the element is present
+        if (!isPresent(waitFor)) {
+            return false;
+        }
+        // perform the check
+        boolean isDisplayed = getElement().is().displayed();
+        // record the result
+        if (!isDisplayed) {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + NOTDISPLAYED, timeTook, Success.PASS);
+        } else {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + DISPLAYED, timeTook, Success.FAIL);
+        }
+        return !isDisplayed;
+    }
+
+    /**
+     * Verifies that the element is checked. If the element isn't present,
+     * it waits up to the default time (5 seconds) for the element, before
+     * marking this verification as a failure. This information will be logged
+     * and recorded, with a screenshot for traceability and added debugging
+     * support.
+     */
+    void checked();
+
+    default boolean checkChecked(double waitFor, double timeTook) {
+        // record the action
+        getOutputFile().recordAction(getElement().prettyOutput() + CHECKED, waitFor);
+        // verify the element is present
+        if (!isPresent(waitFor)) {
+            return false;
+        }
+        // perform the check
+        boolean isChecked = getElement().is().checked();
+        // record the result
+        if (!isChecked) {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + NOTCHECKED, timeTook, Success.FAIL);
+        } else {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + CHECKED, timeTook, Success.PASS);
+        }
+        return isChecked;
+    }
+
+    /**
+     * Verifies that the element is not checked. If the element isn't present,
+     * it waits up to the default time (5 seconds) for the element, before
+     * marking this verification as a failure. This information will be logged
+     * and recorded, with a screenshot for traceability and added debugging
+     * support.
+     */
+    void notChecked();
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Rename method "notChecked" to prevent any misunderstanding/clash with field "NOTCHECKED" defined on line 45. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS1845)